### ISSUE TITLE
constant evaluation error not produced when generating metadata (cargo check)

### ISSUE THREAD

**Issue by ehuss** [rust-lang/rust#48850]
The following will produce an error when you compile it.  However, if you run `cargo check`, it does not produce any errors or warnings.

```rust
const FOO: u8 = [0][1];

fn main() {
    println!("{:?}", FOO);
}
```

I would think that `cargo check` would produce the same errors (up to linking) as build does.  Does anyone have any insight why the constant evaluation seems to be skipped in this case? 

(Tested with rustc 1.26.0-nightly (2789b067d 2018-03-06))

**Comment by oli-obk** [comment]
I think `cargo check` only compiles up to translation. And it looks like the constant is not actually touched by the compiler until translation.

**Comment by oli-obk** [comment]
closing in favour of the more general https://github.com/rust-lang/rust/issues/49292
